<launch>
  <arg name="voxel_size" default="0.005"/>

  <!-- Download the dataset here: https://projects.asl.ethz.ch/datasets/doku.php?id=iros2017 -->

   <node name="voxblox_node" pkg="voxblox_ros" type="tsdf_server" output="screen" args="-alsologtostderr" clear_params="true">
    <remap from="pointcloud" to="/lrs_filter/common"/>
        <!--<param name="bag_file" value="/home/cedricp/tmp/2023-12-05-21-32-56.bag"/>-->
    <param name="bag_file" value="/home/cedricp/tmp/2023-12-07-16-50-28.bag"/>
    <param name="bag_topic" value="/lrs_filter/common"/>
    <param name="tsdf_voxel_size" value="$(arg voxel_size)" />
    <param name="tsdf_voxels_per_side" value="16" />
    <param name="voxel_carving_enabled" value="true" />
    <param name="color_mode" value="color" />
    <param name="use_tf_transforms" value="true" />
    <param name="timestamp_tolerance_sec" value="0.1" />
    <param name="update_mesh_every_n_sec" value="5.0" />
    <param name="min_time_between_msgs_sec" value="0.0" />
    <param name="max_ray_length_m" value="0.8" />
    <param name="method" value="fast" />
    <param name="use_const_weight" value="false" />
    <param name="allow_clear" value="true" />
    <param name="verbose" value="true" />
    <param name="world_frame" value="cubeur_origin" />
    <param name="pointcloud_queue_size" value="50" />
    <param name="mesh_filename" value="/tmp/voxblox.ply"/>
    <rosparam file="$(find voxblox_ros)/cfg/zero.yaml"/>
  </node>

</launch>
